feature-libs/commerce-quotes/core/facade/commerce-quotes.service.ts

```diff
@@ -256,7 +282,19 @@ export class CommerceQuotesService implements CommerceQuotesFacade {
     return this.performQuoteActionCommand.execute({ quoteCode, quoteAction });
   }
 
-  getQuoteDetails(): Observable<Quote | undefined> {
-    return this.quoteDetailsState$.get();
+  requote(quoteCode: string): Observable<Quote> {
+    return this.requoteCommand.execute({ quoteStarter: { quoteCode } });
+  }
+
+  getQuoteDetails(): Observable<QueryState<Quote | undefined>> {
+    return combineLatest([
+      this.isLoading$,
+      this.quoteDetailsState$.getState(),
+    ]).pipe(
```

Comment:
I think instead of combining an additional/external flag, maybe we can put the filter into the pipe so we pass data only if loading for the state is false?

Reply:
It wouldn't solve my issue and I think that we shouldn't filter data that way (otherwise the `loading` flag would not make sense and couldn't be used in template to show loading spinner in different scenarios than mine). I put additional flag because action like requote took some time and only after that details are reloaded, so in the meantime user is not informed that something is happening under the hood. As you mentioned in comment before, this should be moved to component service.

Reply:
If we want to inform users about stuff happening in the background, using spinner component, I would take path with adding such flag on component level, not service

Reply:
"As you mentioned in comment before, this should be moved to component service." - Paweł Fraś 😂

Reply:
Finally it stays after discussion we had. Flag is necessary for all CMS-driven components.